ui/pages/add-token/add-token.component.js

```diff
@@ -167,7 +165,9 @@ class AddToken extends Component {
       autoFilled: false,
     });
 
-    const addressIsValid = isValidAddress(customAddress);
+    const addressIsValid = isValidHexAddress(customAddress, {
```

Comment:
This was using the `isValidAddress` function from our util file, so we'd want to take the 'B' strategy that you listed (check burn address and set `mixedCaseUseChecksum` to `true`).

Reply:
https://github.com/MetaMask/metamask-extension/blob/v9.4.0/ui/app/pages/add-token/add-token.component.js#L164

Reply:
Ah, I didn't realize we switched between those two methods during the `ethereumjs-util` update. Good catch!